kernel/src/rvm/mod.rs

```diff
@@ -38,18 +38,55 @@ mod rvm_extern_fn {
     fn rvm_dealloc_frame(paddr: usize) {
         dealloc_frame(paddr)
     }
+    #[rvm::extern_fn(alloc_frame_x4)]
+    fn rvm_alloc_frame_x4() -> Option<usize> {
+        use crate::memory::alloc_frame_contiguous;
+        alloc_frame_contiguous(4, 2)
+    }
+
+    #[rvm::extern_fn(dealloc_frame_x4)]
+    fn rvm_dealloc_frame_x4(paddr: usize) {
+        dealloc_frame(paddr)
+    }
 
     #[rvm::extern_fn(phys_to_virt)]
     fn rvm_phys_to_virt(paddr: usize) -> usize {
         phys_to_virt(paddr)
     }
 
     #[cfg(target_arch = "x86_64")]
-    #[rvm::extern_fn(x86_all_traps_handler_addr)]
-    unsafe fn rvm_x86_all_traps_handler_addr() -> usize {
-        extern "C" {
-            fn __alltraps();
-        }
-        __alltraps as usize
+    #[rvm::extern_fn(is_host_timer_interrupt)]
+    fn rvm_x86_is_host_timer_interrupt(_vec: u8) -> bool {
+        // TODO: fill in the blanks.
+        false
+    }
+    #[cfg(target_arch = "x86_64")]
+    #[rvm::extern_fn(is_host_serial_interrupt)]
+    fn rvm_x86_is_host_serial_interrupt(_vec: u8) -> bool {
+        // TODO: fill in the blanks.
+        false
+    }
+
+    #[cfg(any(target_arch = "riscv64", target_arch = "riscv32"))]
+    #[rvm::extern_fn(riscv_trap_handler_no_frame)]
+    fn rvm_riscv_trap_handler_no_frame(sepc: &mut usize) {
+        crate::arch::interrupt::trap_handler_no_frame(sepc);
+    }
+
+    #[cfg(all(
```

Comment:
It would be better to write like this:

```rust
#[cfg(any(target_arch = "riscv64", target_arch = "riscv32"))]
fn rvm_riscv_check_hypervisor_extension() -> bool {
    if cfg!(feature = "hypervisor") {
        true
    } else {
        false
    }    
}
```